volatility3/framework/symbols/windows/extensions/consoles.py

```diff
@@ -0,0 +1,439 @@
+        particular executable."""
+        for alias in self.AliasList.to_list(
+            f"{self.get_symbol_table_name()}{constants.BANG}_ALIAS",
+            "ListEntry",
+        ):
+            yield alias
+
+
+class SCREEN_INFORMATION(objects.StructType):
+    """A Screen Information Structure."""
+
+    @property
+    def ScreenX(self):
+        # 22000 change from an array of pointers to _ROW to an array of _ROW
+        row = self.TextBufferInfo.BufferRows.Rows[0]
+        if hasattr(row, "Row"):
+            return row.Row.RowLength2
+        else:
+            return row.RowLength2
+
+    @property
+    def ScreenY(self):
+        return self.TextBufferInfo.BufferCapacity
+
+    def _truncate_rows(self, rows):
+        """To truncate empty rows at the end, walk the list
+        backwards and get the last non-empty row. Use that
+        row index to splice. Rows are created based on the
+        length given in the ROW structure, so empty rows will
+        be ''."""
+
+        non_empty_index = 0
+        rows_traversed = False
+
+        for index, row in enumerate(reversed(rows)):
+            # the string was created based on the length in the ROW structure so it shouldn't have any bad data
+            if len(row.rstrip()) > 0:
+                non_empty_index = index
+                break
+            rows_traversed = True
+
+        if non_empty_index == 0 and rows_traversed:
+            rows = []
+        else:
+            rows = rows[0 : len(rows) - non_empty_index]
+
+        return rows
+
+    def get_buffer(self, truncate_rows=True, truncate_lines=True):
+        """Get the screen buffer.
+
+        The screen buffer is comprised of the screen's Y
+        coordinate which tells us the number of rows and
+        the X coordinate which tells us the width of each
+        row in characters. Windows 10 17763 changed from
+        a large text buffer to a grid of cells, with each
+        cell containing a single wide character in that
+        cell, stored in a CharRowCell object.
+
+        @param truncate: True if the empty rows at the
+        end (i.e. bottom) of the screen buffer should be
+        supressed.
+        """
+        rows = []
+
+        capacity = self.TextBufferInfo.BufferCapacity
+        start = self.TextBufferInfo.BufferStart
+        buffer_rows = self.TextBufferInfo.BufferRows.dereference()
+        buffer_rows.Rows.count = self.TextBufferInfo.BufferCapacity
+
+        for i in range(capacity):
+            index = (start + i) % capacity
+            row = buffer_rows.Rows[index]
+            if hasattr(row, "Row"):
+                row = row.Row
+            try:
+                text = row.get_text(truncate_lines)
+                rows.append(text)
+            except:
+                break
+
+        if truncate_rows:
+            rows = self._truncate_rows(rows)
+
+        if rows:
+            rows = ["=== START OF BUFFER ==="] + rows + ["=== END OF BUFFER ==="]
```

Comment:
I understand why you might want to clearly delimit this for command line output, but it's surely going to be inconvenient for programatic use of this plugin (or if the data were the put into a web interface where the start and end can easily be determined)?